src/downloader.cpp

```diff
@@ -128,20 +128,20 @@ Downloader::Downloader() :
   mp_aria(new Aria2())
 {
   try {
-    for (auto gid : mp_aria->tellActive()) {
+    for (auto gid : mp_aria->tellWaiting()) {
       m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
       m_knownDownloads[gid]->updateStatus(false);
     }
   } catch (std::exception& e) {
-    std::cerr << "aria2 tellActive failed : " << e.what() << std::endl;
+    std::cerr << "aria2 tellWaiting failed : " << e.what() << std::endl;
   }
   try {
-    for (auto gid : mp_aria->tellWaiting()) {
+    for (auto gid : mp_aria->tellActive()) {
       m_knownDownloads[gid] = std::unique_ptr<Download>(new Download(mp_aria, gid));
       m_knownDownloads[gid]->updateStatus(false);
     }
   } catch (std::exception& e) {
-    std::cerr << "aria2 tellWaiting failed : " << e.what() << std::endl;
+    std::cerr << "aria2 tellActive failed : " << e.what() << std::endl;
```

Comment:
Here reversing the order also inverses the issue - now you can have the same download processed twice. In the second loop you only have to create a new download if it is not already in `m_knownDownloads`.